PR for SRVCOM-3054: Update the Serverless 1.33 release notes (#77215)
openshift/openshift-docs: modules/serverless-rn-1-33-0.adoc

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-33-0_{context}"]
= Red Hat {ServerlessProductName} 1.33

> {ServerlessProductName} 1.33 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes.

Reviewer: ```suggestion
{ServerlessProductName} 1.33 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes:
```
See ISG: https://www.ibm.com/docs/en/ibm-style?topic=punctuation-colons#introductory-text

[id="new-features-1-33-0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.12.
* {ServerlessProductName} now uses Knative Eventing 1.12.
* {ServerlessProductName} now uses Kourier 1.12.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.12.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.12.
* The `kn func` CLI plugin now uses `func` 1.13.

* OpenShift Serverless on ARM64 is now available as Technology Preview.
* The `NamespacedKafka` annotation is now deprecated. Use the standard Kafka broker without data plane isolation instead.

> * When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster. This functionality is available as a developer preview.

Reviewer 2: ```suggestion
* When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster. This functionality is available as a Developer Preview.
```

Author: done

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-33-0_{context}"]
= Red Hat {ServerlessProductName} 1.33

{ServerlessProductName} 1.33 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes.

[id="new-features-1-33-0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.12.
* {ServerlessProductName} now uses Knative Eventing 1.12.
* {ServerlessProductName} now uses Kourier 1.12.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.12.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.12.
* The `kn func` CLI plugin now uses `func` 1.13.

> 

Reviewer 3: Blank line:
```suggestion

```
And a few others below for this list.

Reviewer 4: We visually separate this "template" part with just new component versions from "new features proper" part in all release notes for Serverless, so this is OK. Same for newline buffering of each non-just-component-version release note. I've suggested to remove the double empty line below.

Reviewer 4: @briandooley Don't accept this particular suggestion.

> * OpenShift Serverless on ARM64 is now available as Technology Preview (TP).

Reviewer 3: Just "Technology Preview", per https://redhat-documentation.github.io/supplementary-style-guide/#technology-preview:
```suggestion
* OpenShift Serverless on ARM64 is now available as Technology Preview.
```
or, probably more correct:
```suggestion
* OpenShift Serverless on ARM64 is now available as a Technology Preview feature.
```

Reviewer 4: The upper option ("as Technology Preview") is better, because it's somewhat of a misnomer to call support for an arch a feature.

* The `NamespacedKafka` annotation is now deprecated. Use the standard Kafka broker without data plane isolation instead.

* When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster. This functionality is available as a developer preview.

> 

Reviewer 4: ```suggestion
```

> * You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources (`KafkaSource`). This functionality is available as a technology preview (TP) feature, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.

Reviewer 3: If technically accurate:
```suggestion
* You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources, defined by a `KafkaSource` object. This functionality is available as a technology preview (TP) feature, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.
```
Also "Technology Preview":
```suggestion
* You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources, defined by a `KafkaSource` object. This functionality is available as a Technology Preview feature, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.
```

Reviewer 4: Bottom option is good.

[id="known-issues-1-33-0_{context}"]
== Known issues

> * Due to different mount point permissions, direct upload on a cluster build does not work on IBM zSystems (s390x) and IBM Power (ppc64le).

Reviewer 3: There are some common attributes you could use:
```
// IBM Power
:ibm-power-name: IBM Power(R)
:ibm-power-title: IBM Power
// IBM zSystems
:ibm-z-name: IBM Z(R)
:ibm-z-title: IBM Z
```
```suggestion
* Due to different mount point permissions, direct upload on a cluster build does not work on {ibm-z-name} (s390x) and {ibm-power-name} (ppc64le).
```

* Building and deploying a function using Podman version 4.6 fails with the `invalid pull policy "1"` error.
+

> To work around this issue, use Podman version 4.5.

Reviewer 3: Maybe out of scope for the PR, but I see mostly across the RN assembly it's lowercase "podman", but a few recent entries (and this new one) are using init cap "Podman". Suggest syncing up one way or the other going forward.

Reviewer 4: We'll address this in a future PR, this is too cumbersome to do without the original PR creator.

// Module included in the following assemblies
//
// * about/serverless-release-notes.adoc

:_content-type: REFERENCE
[id="serverless-rn-1-33-0_{context}"]
= Red Hat {ServerlessProductName} 1.33

{ServerlessProductName} 1.33 is now available. New features, updates, and known issues that pertain to {ServerlessProductName} on {ocp-product-title} are included in the following notes.

[id="new-features-1-33-0_{context}"]
== New features

* {ServerlessProductName} now uses Knative Serving 1.12.
* {ServerlessProductName} now uses Knative Eventing 1.12.
* {ServerlessProductName} now uses Kourier 1.12.
* {ServerlessProductName} now uses Knative (`kn`) CLI 1.12.
* {ServerlessProductName} now uses Knative for Apache Kafka 1.12.
* The `kn func` CLI plugin now uses `func` 1.13.

> 

Reviewer 4: ```suggestion

OpenShift Serverless on ARM64 is now available as Technology Preview (TP).
```

> * When specifying Kourier for Ingress and using `DomainMapping`, the TLS for OpenShift Route is set to passthrough, and TLS is handled by the Kourier Gateway. Beginning with Serverless 1.31, you can specify the enabled cipher suites on the side of the Kourier Gateway.
> 

Reviewer 4: ```suggestion
```

* The `NamespacedKafka` annotation is now deprecated. Use the standard Kafka broker without data plane isolation instead.

> * You can now easily discover events available within the cluster.

Reviewer 5: This text is very generic

```suggestion
* When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster.
```

Reviewer 5: I think we should also mention that this is a developer preview feature

Author: Added!

> * When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster. This functionality is available as a developer preview 

Reviewer 6: ```suggestion
* When enabling the automatic `EventType` auto-creation, you can now easily discover events available within the cluster. This functionality is available as a developer preview.
```

> * The `kn` event plugin for the Knative CLI (kn) is now generally available. The `kn` event plugin lets you send events directly from your command line to various destinations, streamlining your development workflow.

Reviewer 7: The GA state isn't yet reached. Here's the milestone that tracks it: https://github.com/knative-extensions/kn-plugin-event/milestone/2 and a Jira epic: https://issues.redhat.com/browse/SRVCLI-391

> * The `kn` event plugin for the Knative CLI (kn) is now generally available. The `kn` event plugin lets you send events directly from your command line to various destinations, streamlining your development workflow.

Reviewer 4: ```suggestion

```

> * The new trigger filters feature is now available as a Technology Preview. The new trigger filters are now enabled by default. It allows users to specify a set of filter expressions, where each expression evaluates to either true or false for each event.
> 

Reviewer 4: ```suggestion
```

> * You can now explore the Knative Eventing monitoring dashboards directly within the *Observe* tab of the Developer view in the OpenShift Developer Console. 
> 

Reviewer 4: ```suggestion
```

> * {FunctionsProductName} now supports Go function container and is now available as a Technology Preview (TP) feature.
> 

Reviewer 4: ```suggestion
```

> * You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources. This functionality is available as a developer preview, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.

Reviewer 5: ```suggestion
* You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing for Apache Kafka sources (`KafkaSource`). This functionality is available as a developer preview, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.
```

Author: Added!

> * You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources (`KafkaSource`). This functionality is available as a developer preview, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.

Reviewer 4: ```suggestion
* You can now use the Custom Metrics Autoscaler Operator to autoscale Knative Eventing sources for Apache Kafka sources (`KafkaSource`). This functionality is available as a technology preview (TP) feature, offering enhanced scalability and efficiency for Kafka-based event sources within Knative Eventing.
```

Reviewer 6: If we have this as TP then we should probably list it in the table as well? Just thinking.

[id="known-issues-1-33-0_{context}"]
== Known issues

* Due to different mount point permissions, direct upload on a cluster build does not work on IBM zSystems (s390x) and IBM Power (ppc64le).

* Building and deploying a function using Podman version 4.6 fails with the `invalid pull policy "1"` error.
+
To work around this issue, use Podman version 4.5.

> * When upgrading from version 1.32.0 to 1.32.1, note that all Pods of Knative Services will restart. Ensure that Pods start correctly by verifying that images can be pulled and other dependencies are met to prevent connection drops.

Reviewer 6: Is this note valid? We're documenting 1.33.0

Reviewer 4: @mgencur I suppose this is covered here because there were no 1.32.1 release notes, and a reader would not look for a `1.32.0-to-1.32.1` note in 1.32.0 release notes: we don't expect post-1.32.0 stuff to be in 1.32.0 notes.

If during a `1.32.0-to-1.33.0` upgrade this is not relevant, we can remove this, just tell me.

Reviewer 6: OK. We should remove this note. The issue was fixed in 1.32.1 and the upgrade from 1.32.1 to 1.33.0 works as expected. This is not an issue anymore for 1.33.

Reviewer 8: @rh-max looks like this feedback was not implemented. I still see line 46 in the PR

> * When upgrading from version 1.32.0 to 1.32.1, note that all Pods of Knative Services will restart. Ensure that Pods start correctly by verifying that images can be pulled and other dependencies are met to prevent connection drops.

Reviewer 4: ```suggestion
```
